File: src/main/java/mvp/BaseContract.java
package mvp;

public interface BaseContract {

    interface IView {

    }

    interface IPresenter<V extends IView> {

        void attachView(V view);

        void detachView();

        void onDestroy();
    }


    interface IInteractor {
        void onDestroy();
    }

    interface IOutput {

    }


}

File: src/main/kotlin/chapter2OOP/mvp/BaseContract.kt
package chapter2OOP.mvp

interface BaseContract {

    interface IView
    interface IPresenter<V : IView> {

        fun attachOutput()
        fun attachView(view: V)
        fun detachView()
        fun onDestroy()
    }

    interface IInteractor {
        fun onDestroy()
    }

    interface IOutput
}